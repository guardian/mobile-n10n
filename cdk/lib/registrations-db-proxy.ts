import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { AppIdentity, GuStack } from '@guardian/cdk/lib/constructs/core';
import { GuVpc, SubnetType } from '@guardian/cdk/lib/constructs/ec2';
import type { App } from 'aws-cdk-lib';
import { Arn, ArnFormat, CfnOutput } from 'aws-cdk-lib';
import { SecurityGroup } from 'aws-cdk-lib/aws-ec2';
import {
	DatabaseInstance,
	DatabaseInstanceEngine,
	DatabaseProxy,
	ProxyTarget,
} from 'aws-cdk-lib/aws-rds';
import { Secret } from 'aws-cdk-lib/aws-secretsmanager';

interface DbProxyStackProps extends GuStackProps {
	appName: string;
	dbHost: string;
	dbName: string;
	dbInstanceId: string;
	dbSecurityGroupId: string;
}

export class RegistrationsDbProxy extends GuStack {
	constructor(scope: App, id: string, props: DbProxyStackProps) {
		super(scope, id, props);

		const vpc = GuVpc.fromIdParameter(
			this,
			AppIdentity.suffixText({ app: props.appName }, 'VPC'),
		);

		const dbPort = 5432;
		const secretTemplate = {
			engine: 'postgres',
			host: props.dbHost,
			port: dbPort,
			dbname: props.dbName,
			dbInstanceIdentifier: props.dbInstanceId,
		};
		const dbWorkerSecret = new Secret(this, 'RegistrationDbWorkerSecret', {
			secretName: `registrations-db-worker-secret-${props.stage}`,
			description:
				'Secrets for accessing registration database from worker lambdas',
			generateSecretString: {
				secretStringTemplate: JSON.stringify({
					username: 'worker_user',
					...secretTemplate,
				}),
				generateStringKey: 'password',
			},
		});
		const dbCleanerSecret = new Secret(this, 'RegistrationDbCleanerSecret', {
			secretName: `registrations-db-cleaner-secret-${props.stage}`,
			description:
				'Secrets for accessing registration database from cleaner lambdas',
			generateSecretString: {
				secretStringTemplate: JSON.stringify({
					username: 'cleaner_user',
					...secretTemplate,
				}),
				generateStringKey: 'password',
			},
		});

		const dbSecurityGroup = SecurityGroup.fromSecurityGroupId(
			this,
			'registrations-db-security-group',
			props.dbSecurityGroupId,
		);
		const registrationDb = DatabaseInstance.fromDatabaseInstanceAttributes(
			this,
			'registrations-db',
			{
				instanceEndpointAddress: props.dbHost,
				instanceIdentifier: props.dbInstanceId,
				port: dbPort,
				securityGroups: [dbSecurityGroup],
				engine: DatabaseInstanceEngine.POSTGRES,
			},
		);

		const proxy = new DatabaseProxy(this, 'RegistrationsDbProxy', {
			dbProxyName: `registrations-db-proxy-cdk-${props.stage}`,
			proxyTarget: ProxyTarget.fromInstance(registrationDb),
			secrets: [dbWorkerSecret, dbCleanerSecret],
			vpc,
			iamAuth: false,
			maxConnectionsPercent: 90,
			securityGroups: [dbSecurityGroup],
			requireTLS: false,
			vpcSubnets: {
				subnets: GuVpc.subnetsFromParameter(this, {
					type: SubnetType.PRIVATE,
					app: props.appName,
				}),
			},
		});

		const proxyResourceId = Arn.split(
			proxy.dbProxyArn,
			ArnFormat.COLON_RESOURCE_NAME,
		).resourceName;
		if (proxyResourceId != undefined) {
			const grantId = `arn:aws:rds-db:${this.region}:${this.account}:dbuser:${proxyResourceId}/*`;
			new CfnOutput(this, 'RegistrationsDbProxyId', {
				value: grantId,
				description: 'ID of RDS proxy to registrations database',
				exportName: `RegistrationsDbProxyId-${props.stage}`,
			});
		}
	}
}
