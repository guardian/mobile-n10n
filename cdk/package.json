{
  "name": "cdk",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "tsc",
    "test": "jest",
    "format": "prettier --write \"{lib,bin}/**/*.ts\"",
    "lint": "eslint lib/** bin/** --ext .ts --no-error-on-unmatched-pattern",
    "synth": "cdk synth --path-metadata false --version-reporting false",
    "diff": "cdk diff --path-metadata false --version-reporting false"
  },
  "devDependencies": {
    "@guardian/cdk": "45.1.3",
    "@guardian/eslint-config-typescript": "1.0.1",
    "@guardian/prettier": "1.0.0",
    "@types/jest": "^27.5.0",
    "@types/node": "17.0.42",
    "aws-cdk": "2.25.0",
    "aws-cdk-lib": "2.25.0",
    "constructs": "10.1.17",
    "eslint": "^8.16.0",
    "jest": "^27.5.1",
    "prettier": "^2.7.0",
    "source-map-support": "^0.5.20",
    "ts-jest": "^27.1.4",
    "ts-node": "^10.8.1",
    "typescript": "~4.7.2"
  },
  "prettier": "@guardian/prettier",
  "jest": {
    "testMatch": [
      "<rootDir>/lib/**/*.test.ts"
    ],
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "setupFilesAfterEnv": [
      "./jest.setup.js"
    ]
  },
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true,
      "jest": true
    },
    "extends": [
      "@guardian/eslint-config-typescript"
    ],
    "parserOptions": {
      "ecmaVersion": 2020,
      "sourceType": "module"
    },
    "plugins": [
      "@typescript-eslint"
    ],
    "rules": {
      "@typescript-eslint/no-inferrable-types": 0,
      "import/no-namespace": 2
    },
    "ignorePatterns": [
      "**/*.js",
      "node_modules",
      "cdk.out",
      ".eslintrc.js",
      "jest.config.js"
    ]
  }
}