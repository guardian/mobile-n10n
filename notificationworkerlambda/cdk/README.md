# Notification Lambda CDK

This is the CDK for the Cloudformation that describes the notification lambda architecture.

The `cdk.json` file tells the CDK Toolkit how to execute your app.

## Scope

Currently this defines the notification sender lambdas because there are 5 of
these with almost identical configurations.

## Useful commands

 * `npm run build`   compile typescript to js
 * `npm run watch`   watch for changes and compile
 * `npm run test`    perform the jest unit tests
 * `cdk deploy`      deploy this stack to your default AWS account/region
 * `cdk diff`        compare deployed stack with current state
 * `cdk synth`       emits the synthesized CloudFormation template

## Stack dependencies

It's important to note that there is an [Exported
Value](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-stack-exports.html)
which is exported from this stack and consumed by the Harvester stack. This is
because Harvester needs to know the ARNs of the SQS queues that are created here
for each notifcation sender.

The off-shoot of this is that AWS will prevent you from making a change to this
list of ARNs while it is being consumed by the other stack.

Note that this is only a problem *in the very rare case* of when you are making
a change which actually affects this list of ARNs &mdash; e.g. adding a new
lambda or removing an old one &mdash; any other changes should be fine.

If you are making such a change, the deployment will fail (harmlessly, as it
will just be rolled back).

### Deploying if you change the list of SQS queues

In that case, what you should do is, copy the currently exported value (either
via the console or using the following snippet, replacing `CODE` with the
correct stage of course):

``` shell
$ aws cloudformation list-exports | jq -r '.Exports | map(select(.Name == "NotificationSenderWorkerQueueArns-CODE")) | .[0].Value'
```

and then insert this into the harvester cloudformation as a literal value
instead of the import. The simplest way to do this is just to edit the YAML file
and upload it. The diff would look something like this:

``` shell
diff --git a/notificationworkerlambda/harvester-cfn.yaml b/notificationworkerlambda/harvester-cfn.yaml
index 3e4fcbf1..e83c63fa 100644
--- a/notificationworkerlambda/harvester-cfn.yaml
+++ b/notificationworkerlambda/harvester-cfn.yaml
@@ -110,10 +110,11 @@ Resources:
           Statement:
             Effect: Allow
             Action: sqs:*
-            Resource: !Split
-              - ","
-              - !ImportValue
-                'Fn::Sub': 'NotificationSenderWorkerQueueArns-${Stage}'
+            Resource: "arn:aws:sqs:eu-west-1:[...]"
       - PolicyName: VPC
         PolicyDocument:
           Statement:
```

After that, the following command should indicate that this value is no longer
imported, and then you can freely make any neccessary changes:

``` shell
$ aws cloudformation list-imports --export-name='NotificationSenderWorkerQueueArns-CODE'
An error occurred (ValidationError) when calling the ListImports operation: Export 'NotificationSenderWorkerQueueArns-CODE' is not imported by any stack.
```

After the project has been redeployed with riff-raff, the import should be
automatically put back in place.
