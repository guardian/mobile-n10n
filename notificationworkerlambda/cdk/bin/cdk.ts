import 'source-map-support/register';
import * as cdk from '@aws-cdk/core';
import { SenderWorkerStack } from '../lib/senderworker';

const app = new cdk.App()

new SenderWorkerStack(app, 'SenderWorkerStack', {
  stack: "mobile-notifications-workers",
  stage: "CODE"
})

new SenderWorkerStack(app, 'SenderWorkerStack', {
  stack: "mobile-notifications-workers",
  stage: "PROD"
})
