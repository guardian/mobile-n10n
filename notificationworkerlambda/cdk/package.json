{
  "dependencies": {
    "@guardian/cdk": "^61.3.0",
    "aws-cdk": "2.1007.0",
    "@types/jest": "^29.5.14",
    "aws-cdk-lib": "2.188.0",
    "constructs": "^10.4.2",
    "jest": "^29.7",
    "ts-jest": "^29.3.2",
    "typescript": "^5.8.3"
  },
  "license": "ISC",
  "scripts": {
    "synth": "cdk synth",
    "test": "jest"
  }
}
