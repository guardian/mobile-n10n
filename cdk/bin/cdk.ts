import 'source-map-support/register';
import { App } from 'aws-cdk-lib';
import { RegistrationsDbProxy } from '../lib/registrations-db-proxy';

const app = new App();

export const dbProxyCodeProps = {
	stack: 'mobile-notifications',
	stage: 'CODE',
	appName: 'registrations-db-proxy',
	dbHost:
		'notifications-registrations-db-private-testrig.crwidilr2ofx.eu-west-1.rds.amazonaws.com',
	dbName: 'registrationsCODE',
	dbInstanceId: 'notifications-registrations-db-private-testrig',
	dbSecurityGroupId: 'sg-0a228b61bbf4a0e6a',
};
export const dbProxyProdProps = {
	stack: 'mobile-notifications',
	stage: 'PROD',
	appName: 'registrations-db-proxy',
	dbHost:
		'notifications-registrations-db-private-prod.crwidilr2ofx.eu-west-1.rds.amazonaws.com',
	dbName: 'registrationsPROD',
	dbInstanceId: 'notifications-registrations-db-private-prod',
	dbSecurityGroupId: 'sg-01ee0bddb42ead7e5',
};

new RegistrationsDbProxy(app, 'RegistrationsDbProxy-CODE', dbProxyCodeProps);
new RegistrationsDbProxy(app, 'RegistrationsDbProxy-PROD', dbProxyProdProps);
