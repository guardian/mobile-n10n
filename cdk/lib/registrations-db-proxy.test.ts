import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { RegistrationsDbProxy } from './registrations-db-proxy';
import { dbProxyCodeProps, dbProxyProdProps } from '../bin/cdk';

describe('The RegistrationDbProxy stack', () => {
	it('matches the snapshot on CODE', () => {
		const app = new App();
		const stack = new RegistrationsDbProxy(app, 'RegistrationDbProxy-CODE', dbProxyCodeProps);
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
	it('matches the snapshot on PROD', () => {
		const app = new App();
		const stack = new RegistrationsDbProxy(app, 'RegistrationDbProxy-PROD', dbProxyProdProps);
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
