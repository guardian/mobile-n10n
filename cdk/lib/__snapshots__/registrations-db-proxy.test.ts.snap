// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RegistrationDbProxy stack matches the snapshot on CODE 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuVpcParameter",
      "GuSubnetListParameter",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "RegistrationsDbProxyId": {
      "Description": "ID of RDS proxy to registrations database",
      "Export": {
        "Name": "RegistrationsDbProxyId-CODE",
      },
      "Value": {
        "Fn::Join": [
          "",
          [
            "arn:aws:rds-db:",
            {
              "Ref": "AWS::Region",
            },
            ":",
            {
              "Ref": "AWS::AccountId",
            },
            ":dbuser:",
            {
              "Fn::Select": [
                6,
                {
                  "Fn::Split": [
                    ":",
                    {
                      "Fn::GetAtt": [
                        "RegistrationsDbProxyBAD96AA8",
                        "DBProxyArn",
                      ],
                    },
                  ],
                },
              ],
            },
            "/*",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "VpcId": {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "registrationsdbproxyPrivateSubnets": {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": {
    "RegistrationDbCleanerSecretD579A6EA": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Description": "Secrets for accessing registration database from cleaner lambdas",
        "GenerateSecretString": {
          "GenerateStringKey": "password",
          "SecretStringTemplate": "{"username":"cleaner_user","engine":"postgres","host":"notifications-registrations-db-private-code.crwidilr2ofx.eu-west-1.rds.amazonaws.com","port":5432,"dbname":"registrationsCODE","dbInstanceIdentifier":"notifications-registrations-db-private-code"}",
        },
        "Name": "registrations-db-cleaner-secret-CODE",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "RegistrationDbWorkerSecretD41732AB": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Description": "Secrets for accessing registration database from worker lambdas",
        "GenerateSecretString": {
          "GenerateStringKey": "password",
          "SecretStringTemplate": "{"username":"worker_user","engine":"postgres","host":"notifications-registrations-db-private-code.crwidilr2ofx.eu-west-1.rds.amazonaws.com","port":5432,"dbname":"registrationsCODE","dbInstanceIdentifier":"notifications-registrations-db-private-code"}",
        },
        "Name": "registrations-db-worker-secret-CODE",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "RegistrationsDbProxyBAD96AA8": {
      "Properties": {
        "Auth": [
          {
            "AuthScheme": "SECRETS",
            "IAMAuth": "DISABLED",
            "SecretArn": {
              "Ref": "RegistrationDbWorkerSecretD41732AB",
            },
          },
          {
            "AuthScheme": "SECRETS",
            "IAMAuth": "DISABLED",
            "SecretArn": {
              "Ref": "RegistrationDbCleanerSecretD579A6EA",
            },
          },
        ],
        "DBProxyName": "registrations-db-proxy-cdk-CODE",
        "EngineFamily": "POSTGRESQL",
        "RequireTLS": false,
        "RoleArn": {
          "Fn::GetAtt": [
            "RegistrationsDbProxyIAMRoleF3BC5484",
            "Arn",
          ],
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
        "VpcSecurityGroupIds": [
          "sg-0a228b61bbf4a0e6a",
        ],
        "VpcSubnetIds": {
          "Ref": "registrationsdbproxyPrivateSubnets",
        },
      },
      "Type": "AWS::RDS::DBProxy",
    },
    "RegistrationsDbProxyIAMRoleDefaultPolicyCE03926E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": {
                "Ref": "RegistrationDbWorkerSecretD41732AB",
              },
            },
            {
              "Action": [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": {
                "Ref": "RegistrationDbCleanerSecretD579A6EA",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "RegistrationsDbProxyIAMRoleDefaultPolicyCE03926E",
        "Roles": [
          {
            "Ref": "RegistrationsDbProxyIAMRoleF3BC5484",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "RegistrationsDbProxyIAMRoleF3BC5484": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "rds.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "RegistrationsDbProxyProxyTargetGroupDDA34D50": {
      "Properties": {
        "ConnectionPoolConfigurationInfo": {
          "MaxConnectionsPercent": 90,
        },
        "DBInstanceIdentifiers": [
          "notifications-registrations-db-private-code",
        ],
        "DBProxyName": {
          "Ref": "RegistrationsDbProxyBAD96AA8",
        },
        "TargetGroupName": "default",
      },
      "Type": "AWS::RDS::DBProxyTargetGroup",
    },
    "registrationsdbsecuritygroupfromRegistrationDbProxyCODEregistrationsdbsecuritygroup14CF94DB54323FFA1BF5": {
      "Properties": {
        "Description": "Allow connections to the database Instance from the Proxy",
        "FromPort": 5432,
        "GroupId": "sg-0a228b61bbf4a0e6a",
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": "sg-0a228b61bbf4a0e6a",
        "ToPort": 5432,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
  },
}
`;

exports[`The RegistrationDbProxy stack matches the snapshot on PROD 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuVpcParameter",
      "GuSubnetListParameter",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "RegistrationsDbProxyId": {
      "Description": "ID of RDS proxy to registrations database",
      "Export": {
        "Name": "RegistrationsDbProxyId-PROD",
      },
      "Value": {
        "Fn::Join": [
          "",
          [
            "arn:aws:rds-db:",
            {
              "Ref": "AWS::Region",
            },
            ":",
            {
              "Ref": "AWS::AccountId",
            },
            ":dbuser:",
            {
              "Fn::Select": [
                6,
                {
                  "Fn::Split": [
                    ":",
                    {
                      "Fn::GetAtt": [
                        "RegistrationsDbProxyBAD96AA8",
                        "DBProxyArn",
                      ],
                    },
                  ],
                },
              ],
            },
            "/*",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "VpcId": {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "registrationsdbproxyPrivateSubnets": {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": {
    "RegistrationDbCleanerSecretD579A6EA": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Description": "Secrets for accessing registration database from cleaner lambdas",
        "GenerateSecretString": {
          "GenerateStringKey": "password",
          "SecretStringTemplate": "{"username":"cleaner_user","engine":"postgres","host":"notifications-registrations-db-private-prod.crwidilr2ofx.eu-west-1.rds.amazonaws.com","port":5432,"dbname":"registrationsPROD","dbInstanceIdentifier":"notifications-registrations-db-private-prod"}",
        },
        "Name": "registrations-db-cleaner-secret-PROD",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "RegistrationDbWorkerSecretD41732AB": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Description": "Secrets for accessing registration database from worker lambdas",
        "GenerateSecretString": {
          "GenerateStringKey": "password",
          "SecretStringTemplate": "{"username":"worker_user","engine":"postgres","host":"notifications-registrations-db-private-prod.crwidilr2ofx.eu-west-1.rds.amazonaws.com","port":5432,"dbname":"registrationsPROD","dbInstanceIdentifier":"notifications-registrations-db-private-prod"}",
        },
        "Name": "registrations-db-worker-secret-PROD",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "RegistrationsDbProxyBAD96AA8": {
      "Properties": {
        "Auth": [
          {
            "AuthScheme": "SECRETS",
            "IAMAuth": "DISABLED",
            "SecretArn": {
              "Ref": "RegistrationDbWorkerSecretD41732AB",
            },
          },
          {
            "AuthScheme": "SECRETS",
            "IAMAuth": "DISABLED",
            "SecretArn": {
              "Ref": "RegistrationDbCleanerSecretD579A6EA",
            },
          },
        ],
        "DBProxyName": "registrations-db-proxy-cdk-PROD",
        "EngineFamily": "POSTGRESQL",
        "RequireTLS": false,
        "RoleArn": {
          "Fn::GetAtt": [
            "RegistrationsDbProxyIAMRoleF3BC5484",
            "Arn",
          ],
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
        "VpcSecurityGroupIds": [
          "sg-01ee0bddb42ead7e5",
        ],
        "VpcSubnetIds": {
          "Ref": "registrationsdbproxyPrivateSubnets",
        },
      },
      "Type": "AWS::RDS::DBProxy",
    },
    "RegistrationsDbProxyIAMRoleDefaultPolicyCE03926E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": {
                "Ref": "RegistrationDbWorkerSecretD41732AB",
              },
            },
            {
              "Action": [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": {
                "Ref": "RegistrationDbCleanerSecretD579A6EA",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "RegistrationsDbProxyIAMRoleDefaultPolicyCE03926E",
        "Roles": [
          {
            "Ref": "RegistrationsDbProxyIAMRoleF3BC5484",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "RegistrationsDbProxyIAMRoleF3BC5484": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "rds.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-n10n",
          },
          {
            "Key": "Stack",
            "Value": "mobile-notifications",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "RegistrationsDbProxyProxyTargetGroupDDA34D50": {
      "Properties": {
        "ConnectionPoolConfigurationInfo": {
          "MaxConnectionsPercent": 90,
        },
        "DBInstanceIdentifiers": [
          "notifications-registrations-db-private-prod",
        ],
        "DBProxyName": {
          "Ref": "RegistrationsDbProxyBAD96AA8",
        },
        "TargetGroupName": "default",
      },
      "Type": "AWS::RDS::DBProxyTargetGroup",
    },
    "registrationsdbsecuritygroupfromRegistrationDbProxyPRODregistrationsdbsecuritygroup8D9DEF405432C98C6D17": {
      "Properties": {
        "Description": "Allow connections to the database Instance from the Proxy",
        "FromPort": 5432,
        "GroupId": "sg-01ee0bddb42ead7e5",
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": "sg-01ee0bddb42ead7e5",
        "ToPort": 5432,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
  },
}
`;
